src/core/server/http/base_path_service.ts

```diff
@@ -34,10 +34,19 @@ export class BasePath {
    * See {@link BasePath.get} for getting the basePath value for a specific request
    */
   public readonly serverBasePath: string;
+  /**
+   * The server's publicly exposed base URL, if configured. Includes protocol, host, port (optional) and the
+   * {@link BasePath.serverBasePath}.
+   *
+   * @remarks
+   * Should be used for generating external URL links back to this Kibana instance.
+   */
+  public readonly publicBaseUrl?: string;
 
   /** @internal */
-  constructor(serverBasePath: string = '') {
+  constructor(serverBasePath: string = '', publicBaseUrl?: string) {
     this.serverBasePath = serverBasePath;
+    this.publicBaseUrl = publicBaseUrl;
   }
```

Comment:
I'm wondering if the integrity check between `basePath` and `publicBaseUrl`'s path should be done here instead of `src/core/server/http/http_config.ts`. I guess by doing that in the config, we fail a little faster, but OTOH we can (programmatically) instantiate the service in an invalid state. Probably fine as it is right now.

Reply:
I think for consistency it's best we do validation in a single place. If we want to make `HttpConfig` more robust, we should switch to a pattern where all validation is done in the constructor.